src/uct/cuda/base/cuda_md.c

```diff
@@ -101,6 +101,37 @@ uct_cuda_base_query_md_resources(uct_component_t *component,
                                            num_resources_p);
 }
 
+int uct_cuda_base_is_coherent()
+{
+#if HAVE_CUDA_FABRIC
+    static int coherent = -1;
+    CUdevice cu_device;
+    ucs_status_t status;
+
+    if (coherent != -1) {
+        goto out;
+    }
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuDeviceGet(&cu_device, 0));
+    if (status != UCS_OK) {
+        coherent = 0;
+        goto out;
+    }
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(
```

Comment:
why are we checking CU_DEVICE_ATTRIBUTE_PAGEABLE_MEMORY_ACCESS_USES_HOST_PAGE_TABLES for MNNVL? can you pls add doc?
Is it really needed given that we also check NVLINKS state?

Reply:
took it from the existing logic. 
@Akshay-Venkatesh, can you pls comment? Do we really need this check?

Reply:
This is because MNNVL is supported only on coherent platforms and the above device attribute is a check for coherent platforms. If this is not true, then other checks can be completely skipped.